src/renderer/modals/newServer/newServer.tsx

```diff
@@ -25,14 +26,38 @@ const onSave = (data: TeamWithIndex) => {
     window.postMessage({type: MODAL_RESULT, data}, window.location.href);
 };
 
-const start = async () => {
-    ReactDOM.render(
+const NewServerModalWrapper: React.FC = () => {
+    const [unremoveable, setUnremovable] = useState<boolean>();
+
+    const handleNewServerMessage = (event: {data: ModalMessage<boolean>}) => {
+        switch (event.data.type) {
+        case MODAL_UNCLOSEABLE: {
+            setUnremovable(event.data.data);
+            break;
+        }
+        default:
+            break;
+        }
+    };
+
+    useEffect(() => {
+        window.addEventListener('message', handleNewServerMessage);
```

Comment:
Don't we need to remove this event listener at some point?

Reply:
Based on the comment below, we don't NEED to, but it is a best practice so I'll remove it.